src/server/auth/tokenx/index.ts

```diff
@@ -0,0 +1,28 @@
+import serverLogger from "@/server/utils/serverLogger";
+import { ApiErrorException, generalError } from "@/common/api/axios/errors";
+import { grant } from "./tokenx.grant";
+
+export async function getTokenX(
+  subjectToken: string,
+  audience: string
+): Promise<string> {
+  let tokenX;
+
+  try {
+    tokenX = await grant(subjectToken, audience);
+  } catch (e) {
+    serverLogger.error(
+      `Failed grant for client id: ${audience}. Error message: ${e}`
+    );
+    throw new ApiErrorException(generalError(new Error("Failed grant")));
```

Comment:
hm, vil denne informasjonen "lekke" ut til frontend nå? Tenker frontend ikke bør få vite om f.eks "failed grant" for å unngå å gi noe inside info til haxxors

Reply:
Dersom grant er feil bør vi vel heller gi http 401 og gå til login, tror jeg?

Reply:
Jeg tror Next-connect error handler plukker error'en og kaster videre den generelle feilmeldingen: "Det skjedde en uventet feil. Vennligst prøv igjen senere.". Checkmate h4xx0rz.

Siden feilene kommer fra idporten/wonderwall er jeg redd det blir en evig loop om vi prøver å gå til login igjen?